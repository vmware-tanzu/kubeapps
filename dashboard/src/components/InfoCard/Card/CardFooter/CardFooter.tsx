// Copyright 2020-2023 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import React from "react";

export interface ICardFooterProps {
  children: React.ReactNode;
}

const CardFooter = ({ children }: ICardFooterProps) => {
  return <footer className="card-footer">{children}</footer>;
};

export default CardFooter;
