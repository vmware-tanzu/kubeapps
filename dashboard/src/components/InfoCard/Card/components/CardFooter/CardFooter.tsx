// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import PropTypes from "prop-types";
import React from "react";

const CardFooter = ({ children }) => <footer className="card-footer">{children}</footer>;

CardFooter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
  ]).isRequired,
};

export default CardFooter;
