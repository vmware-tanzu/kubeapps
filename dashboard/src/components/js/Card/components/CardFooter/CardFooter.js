import PropTypes from "prop-types";
import React from "react";

const CardFooter = ({ children }) => <footer className="card-footer">{children}</footer>;

CardFooter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
  ]).isRequired,
};

export default CardFooter;
