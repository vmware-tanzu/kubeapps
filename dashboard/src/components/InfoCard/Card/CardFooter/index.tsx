// Copyright 2020-2023 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import CardFooter from "./CardFooter";
export default CardFooter;
