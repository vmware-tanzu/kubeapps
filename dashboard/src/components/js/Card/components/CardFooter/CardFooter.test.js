// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { shallow } from "enzyme";
import React from "react";
import CardFooter from ".";

describe(CardFooter, () => {
  it("renders the content correctly", () => {
    const text = "My Text";
    const wrapper = shallow(<CardFooter>{text}</CardFooter>);

    expect(wrapper).toHaveText(text);
    expect(wrapper).toMatchSnapshot();
  });

  it("includes the expected CSS class", () => {
    const wrapper = shallow(<CardFooter>Test</CardFooter>);
    expect(wrapper).toHaveClassName("card-footer");
  });
});
