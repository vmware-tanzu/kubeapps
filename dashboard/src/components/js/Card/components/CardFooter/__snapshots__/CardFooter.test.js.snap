// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter renders the content correctly 1`] = `
<footer
  className="card-footer"
>
  My Text
</footer>
`;
